### ISSUE TITLE
Newsfeed Widget

### ISSUE THREAD

**Issue by cruux13** [spacecowboy/Feeder#384]
### Checklist

- [X] I have used the search function for [**OPEN**](https://github.com/spacecowboy/feeder/issues) issues to see if someone else has already submitted the same feature request.
- [X] I have **also** used the search function for [**CLOSED**](https://github.com/spacecowboy/feeder/issues?q=is%3Aissue+is%3Aclosed) issues to see if the feature was already implemented and is just waiting to be released, or if the feature was rejected.
- [X] I will describe the problem with as much detail as possible.
- [X] This request contains only one single feature, **not** a list of multiple (related) features.

### App version

2.6.33

### Where did you get the app from

F-Droid

### Feature or improvement you want

Large customizable widget to emulate similar experience as with Google News Feed

### Why should this be added?

For people who got used to Google News Feed Swipe on start page as me, it would finally make me uninstall Google News Feed and only use Feeder as my main News source. Have seen lots of requests on reddit from people searching for a proper Google News Feed alternative. Feeder was mentioned a lot but it unfortunately has no widget support so people did not find a solution with Feeder.

### Screenshots / Drawings / Technical details

UI should be similar to Google Newsfeed. Small Banners you can scroll through. When clicking on an article, the feedly app or a Browser Tab should open with full details of the article / post.

**Comment by MatthewTighe** [comment]
Hey there! I'm a professional Android developer considering contributing to this feature. I'm not immediately finding a contribution guide. Are you accepting contributions? Can I submit an implementation without official UX?

**Comment by spacecowboy** [comment]
Hi @MatthewTighe , sorry for the delay.

Yes, contributions are welcome.

I'd appreciate it if you could show what you're thinking about in terms of UI before you go and spend a lot of time building it. Because besides the actual Widget itself, the more important UI to me is the configuration UI.

Somewhere, it must be possible to configure the widget with what feed/tag it should display. At least in my head. And that requires some thought.

**Comment by MatthewTighe** [comment]
I'm glad you bring that up! I had been using inoreader for a bit and my main UX friction was that I felt I was missing a drop-down to swap between feeds directly in the widget UI. I'd imagine it being selectable where "newsfeed" text is in the shot below. Other than that I found it to be pretty much was I was looking for, so I'd probably start by trying to recreate something similar

![Screenshot_20251016-221452.png](https://github.com/user-attachments/assets/4a96b6c3-1ea3-4f17-94e3-3efb6b74668e)

**Comment by spacecowboy** [comment]
@MatthewTighe well you should know that Android Widgets are pretty constrained in what they can do. Opening a dropdown is not possible in a widget.

But pressing the cogs icon to open a widget configuration Activity is possible.

**Comment by MatthewTighe** [comment]
Ah, I haven't actually built a widget yet but that makes sense. RIP dream of a convenient dropdown haha. Looking at the [canonical widgets page](https://developer.android.com/design/ui/mobile/guides/widgets/layouts) perhaps a second iteration of this could make some top feeds available as buttons at the top or something?

Here's what the configuration activity looks like for Inoreader. Would something similar work, using the  list that exists in the Feeder sidebar currently?
    
![Screenshot_20251022-101105.png](https://github.com/user-attachments/assets/6e76dc75-7d75-48dc-a5a6-8fe2df6eb9b3)

**Comment by spacecowboy** [comment]
Yes, using the list in the navigation drawer sounds absolutely correct. Great if one could make the same choices too (which is either a feed tag, string, or a feed id, long)

The logic in the app is that given the currently selected Pair of Tag and FeedId, 

1. If tag is not empty, then show the specified tag
2. If feedId is greater than 0, show the specified feed
3. else show all 


I think for the widget, it might make sense to always show read items. At least in first version.

**Comment by MatthewTighe** [comment]
That all sounds good to me! I think I understand enough to put a patch together - are you feeling that we are aligned on the UX direction?

**Comment by spacecowboy** [comment]
Sounds good to me

**Comment by MatthewTighe** [comment]
👋 Just checking in! I'm still working on this but it's massively slow going. Busy season for me, and haven't historically done much programming in my free time. That said, thought I'd drop in a sneak preview of how it's going so far

https://github.com/user-attachments/assets/93303590-0400-4961-b735-526c15d06831

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi there can i leave some feedback on the widget here or would you like me to open up new issues for it?

**Comment by spacecowboy** [comment]
Best to open new issues for bugs please

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Im sorry, i didnt clarify. The feedback are more improvement points not bugs but ill open up  seperate issues for it

**Comment by spacecowboy** [comment]
If there are feature requests or bugs, please open issues.

If you just want to ask questions or talk about something, please open a discussion topic: https://github.com/spacecowboy/Feeder/discussions

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Opened issues already for everything i encountered :)
